core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -887,10 +887,15 @@ class ReplicaManager(val config: KafkaConfig,
       val partitionFetchSize = fetchInfo.maxBytes
       val followerLogStartOffset = fetchInfo.logStartOffset
 
-      brokerTopicStats.topicStats(tp.topic).totalFetchRequestRate.mark()
-      brokerTopicStats.allTopicsStats.totalFetchRequestRate.mark()
-
       try {
+        if (allPartitions.contains(tp)) {
+           brokerTopicStats.topicStats(tp.topic).totalFetchRequestRate.mark()
+           brokerTopicStats.allTopicsStats.totalFetchRequestRate.mark()
```

Comment:
It seems that we can always call this one. It's only the topic variant that should not be called if the topic has been deleted.

Reply:
agreed